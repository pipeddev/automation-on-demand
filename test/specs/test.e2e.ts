import { expect } from "@wdio/globals";
import LoginPage from "../pageobjects/login.page.js";
import SecurePage from "../pageobjects/secure.page.js";

describe("My Login application", () => {
  it("should login with valid credentials", async () => {
    await LoginPage.open();

    await LoginPage.login("tomsmith", "SuperSecretPassword!");
    await expect(SecurePage.flashAlert).toBeExisting();
    await expect(SecurePage.flashAlert).toHaveText(
      expect.stringContaining("You logged into a secure area!")
    );
    await expect(SecurePage.flashAlert).toMatchSnapshot("flashAlert");
  });
});

describe("My Login application Error", () => {
  it("should login with valid credentials fail", async () => {
    await LoginPage.open();

    await LoginPage.login("tomsmith", "SuperSecretPasswor!");
    await expect(SecurePage.flashAlert).toBeExisting();
    await expect(SecurePage.flashAlert).toHaveText(
      expect.stringContaining("You logged into a secure area!")
    );
    await expect(SecurePage.flashAlert).toBeDisplayed();
  });
});
