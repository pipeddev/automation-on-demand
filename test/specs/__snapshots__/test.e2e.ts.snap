// Snapshot v1

exports[`My Login application > should login with valid credentials > flashAlert 1`] = `
"<div data-alert="" id="flash" class="flash success">You logged into a secure area!
  <a href="#" class="close">Ã—</a>
</div>"
`;
